editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/maven/CamelProjectConfigurator.java

```diff
@@ -182,7 +182,10 @@ private void repairCamelFacet(ProjectConfigurationRequest request, IProgressMoni
 		IFacetedProject fproj = ProjectFacetsManager.create(project);
 		if (fproj != null) {
 			IProjectFacetVersion oldfv = fproj.getInstalledVersion(camelFacet);
-			fproj.uninstallProjectFacet(oldfv, null, subMon.split(1));
+			if (oldfv != null) {
```

Comment:
how can it be null? it means that the previous steps were in failure?

Reply:
it can be null if the facet wasn't installed at all.